### PR TITLE
IDEMPIERE-5146 Load/Import Bank Statement issues

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5146

- Add org.compiere.model.MBankStatementLoader.getLastSavedLine() to allow further processing of the saved I_BankStatement record on plugins
- Link Load Bank Statement end of process with Import Bank Statement window (allow addLog in process with Record_ID=0 to open directly the window without query)
- Increase size of columns:
  - I_BankStatement.EftMemo -> 4000
  - I_BankStatement.Memo -> 4000
  - C_BankStatementLine.EftMemo -> 4000
  - C_BankStatementLine.Memo -> 4000
  - I_BankStatement.LineDescription -> 1000
  - C_BankStatementLine.Description -> 1000
  - I_BankStatement.EftTrxType -> 255
  - C_BankStatementLine.EftTrxType -> 255
  - I_BankStatement.EftReference -> 255
  - C_BankStatementLine.EftReference -> 255
  - I_BankStatement.ReferenceNo -> 255
  - C_BankStatementLine.ReferenceNo -> 255

### CODE REVIEW

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java [hunk 1]
```diff
@@ -449,7 +449,7 @@ private void appendRecordLogInfo(ProcessInfoLog[] m_logs, HtmlBasedComponent inf
 			if (msgPresents) {
 				Td td = new Td();
 				if (log.getP_Msg() != null) {
-					if (log.getAD_Table_ID() > 0 && log.getRecord_ID() > 0) {
+					if (log.getAD_Table_ID() > 0) {
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java:452]: this is a change of behavior that's not fully backward compatible.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/StatusBar.java [hunk 2]
```diff
@@ -207,8 +207,7 @@ public void setStatusLine (String text, boolean error, ProcessInfoLog[] m_logs)
 			div = new Div();
 			for (int i = 0; i < m_logs.length; i++) {
 				if (m_logs[i].getP_Msg() != null) {
-					if (m_logs[i].getAD_Table_ID() > 0
-							&& m_logs[i].getRecord_ID() > 0) {
+					if (m_logs[i].getAD_Table_ID() > 0) {
```
[the remark this answers is not in the document]
Comment (CarlosRuiz-globalqss) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/StatusBar.java:210]: The change on AEnv.java controls this case, if the record id is 0 then it just open the window with no query.

I added this to allow showing a link to _Import Bank Statement_ at the end of the _Load Bank Statement_ process.

But is not a big deal, just presentation/usability, if you prefer I can split this piece of the pull request in a ticket specifically for that.

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java (hunk 1 above)
Comment (CarlosRuiz-globalqss) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessDialog.java:452]: same as above - managed with the change on AEnv

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/StatusBar.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (hengsin) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/StatusBar.java:210]: alright, it is really subjective though whether this change is an improvement in user experience.

[the remark this answers is not in the document]
Comment (CarlosRuiz-globalqss) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/adwindow/StatusBar.java:210]: IMO, showing the next step of the process is usability, if is shown with a link, even better - that was the point of IDEMPIERE-370.

So, better to move this code to its own ticket?

**Comment by CarlosRuiz-globalqss** [comment]
I reverted the changes related to:
Link Load Bank Statement end of process with Import Bank Statement window (allow addLog in process with Record_ID=0 to open directly the window without query)
will do that in a different ticket

**Pull request merged**